src/workerd/api/crypto/rsa.c++

```diff
@@ -225,6 +253,196 @@ SubtleCrypto::JsonWebKey Rsa::toJwk(KeyType keyType,
   return jwk;
 }
 
+kj::Maybe<AsymmetricKeyData> Rsa::fromJwk(KeyType keyType,
+                                          const SubtleCrypto::JsonWebKey& jwk) {
+  ClearErrorOnReturn clearErrorOnReturn;
+
+  if (jwk.kty != "RSA"_kj) return kj::none;
+  auto n = JSG_REQUIRE_NONNULL(jwk.n.map([](auto& str) { return str.asPtr(); }), Error,
+                               "Invalid RSA key in JSON Web Key; missing or invalid "
+                               "Modulus parameter (\"n\").");
+  auto e = JSG_REQUIRE_NONNULL(jwk.e.map([](auto& str) { return str.asPtr(); }), Error,
+                               "Invalid RSA key in JSON Web Key; missing or invalid "
+                               "Exponent parameter (\"e\").");
+
+  auto rsa = OSSL_NEW(RSA);
+
+  static constexpr auto kInvalidBase64Error = "Invalid RSA key in JSON Web Key; invalid base64."_kj;
+
+  // TODO(now): Use a decodeBase64Url variant
+  auto nDecoded = toBignumUnowned(simdutfBase64UrlDecodeChecked(n, kInvalidBase64Error));
+  auto eDecoded = toBignumUnowned(simdutfBase64UrlDecodeChecked(e, kInvalidBase64Error));
+  JSG_REQUIRE(RSA_set0_key(rsa.get(), nDecoded, eDecoded, nullptr) == 1, Error,
+              "Invalid RSA key in JSON Web Key; failed to set key parameters");
+
+  if (keyType == KeyType::PRIVATE) {
+    auto d = JSG_REQUIRE_NONNULL(jwk.d.map([](auto& str) { return str.asPtr(); }), Error,
+                                 "Invalid RSA key in JSON Web Key; missing or invalid "
+                                 "Private Exponent parameter (\"d\").");
+    auto p = JSG_REQUIRE_NONNULL(jwk.p.map([](auto& str) { return str.asPtr(); }), Error,
+                                 "Invalid RSA key in JSON Web Key; missing or invalid "
+                                 "First Prime Factor parameter (\"p\").");
+    auto q = JSG_REQUIRE_NONNULL(jwk.q.map([](auto& str) { return str.asPtr(); }), Error,
+                                 "Invalid RSA key in JSON Web Key; missing or invalid "
+                                 "Second Prime Factor parameter (\"q\").");
+    auto dp = JSG_REQUIRE_NONNULL(jwk.dp.map([](auto& str) { return str.asPtr(); }), Error,
```

Comment:
Are the CRT parameters required for a JWK RSA key to be valid? I assumed that "plain" RSA keys are also supported.

Reply:
May not be strictly required for some cases. Will adjust this as necessary when I start working on the tests and verifying that we're matching node.js' behavior correctly.